src/transformStateWithClones.js

```diff
@@ -7,7 +7,49 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const clones = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        addProperties(stateClone, extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateClone, keysToRemove);
+        break;
+
+      case 'clear':
+        clear(stateClone);
+        break;
+
+      default:
+        return;
```

Comment:
checklist.md:
3.[CODE STYLE]: switch/case should always have default case for error handling.